spring-vault-core/src/main/java/org/springframework/vault/authentication/AwsEc2Authentication.java

```diff
@@ -184,8 +186,8 @@ private VaultToken createTokenUsingAwsEc2() {
 			return LoginTokenUtil.from(response.getAuth());
 		}
 		catch (HttpStatusCodeException e) {
-			throw new VaultException(String.format("Cannot login using AWS-EC2: %s",
-					VaultResponses.getError(e.getResponseBodyAsString())));
+			throw new VaultHttpException(String.format("Cannot login using AWS-EC2: %s",
+					VaultResponses.getError(e.getResponseBodyAsString())), e.getStatusCode());
```

Comment:
In some sense, `VaultHttpException: Cannot obtain Identity Document from` is the cause for a `VaultLoginException`

Reply:
Your point?

Reply:
This comment requires some context from `createTokenUsingAwsEc2()`: We're throwing `VaultHttpException` in `createTokenUsingAwsEc2()`. It bubbles up to the user without the context in which it happened (Login through AWS-EC2). It would make sense to catch runtime exceptions and wrap these into `VaultLoginException`.

Login failure handling become much more evident as it's a recurring pattern.